README.md

```diff
@@ -19,6 +19,148 @@ It is *highly* recommended that users of the library learn [Google Protocol Buff
+* `attempt_implicit_secure` indicates whether to attempt to immediately connect to the network element, acquire whatever certificate is available, and set up an encrypted channel utilizing the retrieved certificate. This is still insecure, but encrypted and useful for testing TLS functionality.
+
+```python
+as_secure(
+    self,
+    root_certificates=None,
+    private_key=None,
+    certificate_chain=None,
+    root_from_target=False,
+    target_name_from_root=False,
+    channel_options=None,
+    from_file=False
+)
+```
+
+* `root_certificates`, `private_key`, and `certificate_chain` all simply just wrap [`grpc.ssl_channel_credentials`](https://grpc.github.io/grpc/python/grpc.html#grpc.ssl_channel_credentials) and correspond directly.
+* `root_from_target` indicates to download the root certificate from the target network element.
+* `target_name_from_root` attempts to extract the `grpc.ssl_target_name_override` option with the first CN attribute in the root certificate. This effectively allows us to skip the server name verification of the certificate. This functionality mimics a subset of Go's `TLS.InsecureSkipVerify` which leaves TLS susceptible to MITM attacks - only use for testing.
+* `channel_options` is an array of gRPC option tuples, such as `[("grpc.ssl_target_name_override", "ems.cisco.com")]`. This assumes some knowledge of gRPC options.
+* `from_file` indicates that the `root_certificates`, `private_key`, and `certificate_chain` variables represent file paths which should be read from the filesystem as opposed to the certificate content being passed directly in the variables.
+
+```python
+as_insecure(
+    self,
+    channel_options=None,
+    compression=None
+)
+```
+
+* `as_insecure` utilizes an insecure gRPC channel and is explicitly against gNMI specification. This option is not recommended and exists purely due to insecure server implementations existing. Utilizing `as_secure(root_from_target=True, target_name_from_root=True)` should negate most necessity or desire for this option.
+
+```python
+with_authentication(
+    self,
+    username,
+    password
+)
+```
+
+* This method simply sets the `username` and `password` to utilize for authenticating RPCs with the devices as metadata (Call credentials).
+
+#### Initialization Examples
+
+Using an encrypted channel automatically getting the certificate from the device, quick for testing:
+
+```python
+from cisco_gnmi import Client
+
+client = Client(
+    '127.0.0.1:9339',
+    attempt_implicit_secure=True
+).with_authentication(
+    'admin',
+    'its_a_secret'
+)
+
+# More explicitly...
+
+client = Client(
+    '127.0.0.1:9339'
+).as_secure(
+    root_from_target=True,
+    target_name_from_root=True
+).with_authentication(
+    'admin',
+    'its_a_secret'
+)
+```
+
+
+Using an owned root certificate on the filesystem:
+
+```python
+from cisco_gnmi import Client
+
+client = Client(
+    '127.0.0.1:9339'
+).as_secure(
+    'ems.pem',
```

Comment:
In this example, how is the CA determined?

Reply:
I quite honestly do not know. This is a good exercise in security.